FxEvents/FxEvents.Server/EventSystem/ServerGateway.cs

```diff
@@ -40,12 +40,16 @@ public void Push(string pipeline, int source, byte[] buffer)
                 Events.TriggerAllClientsEvent(pipeline, buffer);
         }
 
+        private int GetSource(Remote source)
+        {
+            return int.Parse(source.ToString().Substring(7, 1));
```

Comment:
what if the source has more than 1 char for its source? like 22 or 115?

Reply:
This, is why I do not want to use Substring, the best solution to this would be using `indexOf("(")`

```cs
int position = source.ToString().IndexOf("(");
if (position < 0)
	throw new Excepation("")
return source.ToString().Substring(position, 1));
```

Reply:
Scrapped this completely after @thorium-cfx comment.